tests/trainer/test_data_loading.py

```diff
@@ -102,3 +102,15 @@ def check_replace_distrubuted_sampler(tmpdir, save_preds_on_dl_idx, accelerator,
 @pytest.mark.parametrize("mode", [1, 2])
 def test_replace_distrubuted_sampler_custom_dataloader_custom_batch_sampler(tmpdir, mode):
     check_replace_distrubuted_sampler(tmpdir, True, "ddp", 2, 2, mode)
+
+
+@RunIf(min_gpus=2)
```

Comment:
Can you mock the number of gpus instead and not run a full test? as we only care about testing the warning

Reply:
When mocking the gpu env, any idea what the exception raised would be? @carmocca